### PR TITLE
Fix issues in ElasticSearch 7.14 and add log cases to ES7.14

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/7425.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/segment/SegmentRecord.java [hunk 1]
```diff
@@ -119,7 +121,9 @@ public String id() {
             map.put(TRACE_ID, storageData.getTraceId());
             map.put(SERVICE_ID, storageData.getServiceId());
             map.put(SERVICE_INSTANCE_ID, storageData.getServiceInstanceId());
-            map.put(ENDPOINT_NAME, storageData.getEndpointName());
+            if (nonNull(storageData.getEndpointName())) {
+                map.put(ENDPOINT_NAME, storageData.getEndpointName());
+            }
             map.put(ENDPOINT_ID, storageData.getEndpointId());
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/segment/SegmentRecord.java:127]: I think this should not be added. Once this has error, the code already has bug. Trace should not have no endpoint.

File: oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/ZipkinSpanRecord.java [hunk 2]
```diff
@@ -120,7 +122,9 @@ public String id() {
             map.put(SPAN_ID, storageData.getSpanId());
             map.put(SERVICE_ID, storageData.getServiceId());
             map.put(SERVICE_INSTANCE_ID, storageData.getServiceInstanceId());
-            map.put(ENDPOINT_NAME, storageData.getEndpointName());
+            if (nonNull(storageData.getEndpointName())) {
+                map.put(ENDPOINT_NAME, storageData.getEndpointName());
+            }
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/ZipkinSpanRecord.java:127]: I think this should not be added. Once this has error, the code already has bug. Trace should not have no endpoint.

**Review by wu-sheng — COMMENTED** [review]
Let's limit the not null in the certain scopes.

**Comment by kezhenxu94** [comment]
> Let's limit the not null in the certain scopes.

The nullability check is added according to whether they are annotated with `org.apache.skywalking.oap.server.core.storage.annotation.Column#matchQuery` that will add a `copy_to` field, which causes the problem. If we skip some of the cases mentioned in the comments, the root cause (`serviceName` being null) is hidden behind the ElasticSearch level message, although `serviceName` is expected to be non null,

**Comment by wu-sheng** [comment]
> > Let's limit the not null in the certain scopes.
> 
> The nullability check is added according to whether they are annotated with `org.apache.skywalking.oap.server.core.storage.annotation.Column#matchQuery` that will add a `copy_to` field, which causes the problem. If we skip some of the cases mentioned in the comments, the root cause (`serviceName` being null) is hidden behind the ElasticSearch level message, although `serviceName` is expected to be non null,

The changes on the log side are reasonable, and should be done. But why we bother to check these `should-not-null` fields? If you want to check and expose the cause, check on the `source#prepare` stage rather than here.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**